hyperparameter_tuning/service.py

```diff
@@ -0,0 +1,124 @@
+from bayes_optuna import optuna_hpo
+
+autotune_object_ids = []
+
+
+class HTTPRequestHandler(BaseHTTPRequestHandler):
+    def _set_response(self, status_code, return_value):
+        # TODO: add status_message
+        self.send_response(status_code)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        self.wfile.write(return_value.encode('utf-8'))
+
+    def do_POST(self):
+        # TODO: perform better regex search
+        if re.search('/experiment_trials', self.path):
+            content_type, params = cgi.parse_header(self.headers.get('content-type'))
+            if content_type == 'application/json':
+                length = int(self.headers.get('content-length'))
+                str_object = self.rfile.read(length).decode('utf8')
+                json_object = json.loads(str_object)
+                # TODO: validate structure of json_object for each operation
+                if json_object["operation"] == "EXP_TRIAL_GENERATE_NEW":
+                    if json_object["id"] not in autotune_object_ids:
+                        autotune_object_ids.append(json_object["id"])
+                        get_search_create_study(json_object["id"], json_object["operation"], json_object["url"])
+                        trial_number = optuna_hpo.TrialDetails.trial_number
+                        self._set_response(200, str(trial_number))
+                    else:
+                        self._set_response(400, "-1")
+                elif json_object["operation"] == "EXP_TRIAL_GENERATE_SUBSEQUENT":
+                    if json_object["id"] in autotune_object_ids:
+                        get_search_create_study(json_object["id"], json_object["operation"], json_object["url"])
+                        trial_number = optuna_hpo.TrialDetails.trial_number
+                        self._set_response(200, str(trial_number))
+                    else:
+                        self._set_response(400, "-1")
+                elif json_object["operation"] == "EXP_TRIAL_RESULT":
+                    if json_object["id"] in autotune_object_ids and json_object["trial_number"] == optuna_hpo.TrialDetails.trial_number:
+                        optuna_hpo.TrialDetails.trial_result = json_object["trial_result"]
+                        optuna_hpo.TrialDetails.result_value_type = json_object["result_value_type"]
+                        optuna_hpo.TrialDetails.result_value = json_object["result_value"]
+                        optuna_hpo.TrialDetails.trial_result_received = 1
+                        self._set_response(200, "0")
+                    else:
+                        self._set_response(400, "-1")
+                else:
+                    self._set_response(400, "-1")
+            else:
+                self._set_response(400, "-1")
+        else:
+            self._set_response(403, "-1")
+
+    def do_GET(self):
+        # TODO: perform better regex search
+        if re.search('/experiment_trials', self.path):
```

Comment:
I'd suggest that you move all of the GET and POST code into separate functions and have only the error handling code here.